### DISCUSSION TITLE
Add <meta name=Date for SEO using mkdocs-git-revision-date-localized-plugin

### DISCUSSION THREAD

**Discussion by xsa** [squidfunk/mkdocs-material#5339] [Q&A] [answered]
Hi everyone,

I'm looking to leverage the _mkdocs-git-revision-date-localized-plugin_ to add the _Date_ `<meta` tag to my pages.

Been trying by adding the below to my `main.html`, in the` {% extrahead %}` block along with the other `<meta` tags, but it does not seem to work. Is there anything I've missed? Or has anyone achieve the same using something different?

`<meta name="Date" scheme="iso8601" content="{{ git_revision_date_localized }}" />`

Thanks in advance.

-X.

**Comment by squidfunk** [comment]
Did you set up [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme), i.e. `custom_dir`? Without knowing exactly what you did, we can only guess.

**Reply by xsa** [reply]
Yes, theme extension is enabled. Everything is working from that perspective, as I'm using other blocks there too.

I also get the Last Updated data on each page, coming from the plugin, e.g. `Last update: April 6, 2023`

**Comment by kamilkrzyskow** [comment, marked as answer]
```html
{%- block extrahead -%}
    {%- if page.meta and page.meta.git_revision_date_localized -%}
        <meta name="Date" scheme="iso8601" content="{{ page.meta.git_revision_date_localized_raw_iso_date }}">
    {%- endif -%}
{%- endblock -%}
```

Like you said in the response to squidfunk, the Last Update date appears, so I checked how the theme handles it:
https://github.com/squidfunk/mkdocs-material/blob/958a8e0568e1c51b7082f45bb77cba10f2ad2b9e/src/partials/source-file.html#L29

but this contained a HTML tag, and the raw date string was needed, so I checked how the plugin stores the `meta` variables:

https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/blob/6150fad41a83abbf7500cf9c75696c5193532305/mkdocs_git_revision_date_localized_plugin/plugin.py#L242-L243

luckily it does provide raw strings to the date.

---

The final result looked like this:
```html
<meta name="Date" scheme="iso8601" content="2023-02-26">
```

However, doing a quick search online, I'm not sure whether or not the `meta name="date"` tag even exists 😅 or if it's made up together with the `scheme` attribute. Finding how to get the raw date was easier than checking this.

**Reply by xsa** [reply]
AH! Pretty neat, it seems to do the trick. I had also looked at the source and how the page would handle it, so I've tried with `{{ page.meta.git_revision_date_localized_raw_iso_date }}` and it was not working.I did not enclose the meta tag in the `if` statement like you though. Thank you very much!

**Reply by kamilkrzyskow** [reply]
Using `page.meta...` without the `if` enclosure, leads to a `page.meta is None` error. This happens because the `404.html` template also gets processed and it has no `meta`. I think that the whole error Traceback properly showcases where the issue originates.
Glad it works for you :v:
